internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -566,18 +566,23 @@ func (r *PreprovisioningImageReconciler) AddIronicAgentToInfraEnv(ctx context.Co
 	return updated, nil
 }
 
-func (r *PreprovisioningImageReconciler) getIPFamilyForInfraEnv(ctx context.Context, infraEnv *common.InfraEnv) (v4 bool, v6 bool, err error) {
-	if infraEnv.ClusterID == "" {
+func (r *PreprovisioningImageReconciler) getIPFamilyForInfraEnv(ctx context.Context, infraEnv *aiv1beta1.InfraEnv, internalInfraEnv *common.InfraEnv) (v4 bool, v6 bool, err error) {
+	ipFamilyAnnotation := infraEnv.GetAnnotations()[infraEnvIPFamilyAnnotation]
+	if ipFamilyAnnotation != "" {
+		families := strings.Split(ipFamilyAnnotation, ",")
+		return funk.Contains(families, ipv4Family), funk.Contains(families, ipv6Family), nil
+	}
+	if internalInfraEnv.ClusterID == "" {
```

Comment:
I changed the name to be more consistent with the other functions.
Now `internalInfraEnv` is the db object and `infraEnv` is the CR.

In this case I needed both because `infraEnv` has the annotations.

Reply:
oh right! I missed that `infraEnv` was referring to the DB object :man_facepalming:, it's a very good change :+1: